packages/clerk-js/src/core/fraudProtection.ts

```diff
@@ -1,29 +1,63 @@
-import { CaptchaChallenge } from '../utils/captcha/CaptchaChallenge';
-import type { Clerk } from './resources/internal';
+import type { CaptchaChallenge } from '../utils/captcha/CaptchaChallenge';
+import type { Clerk, Client } from './resources/internal';
+import { isClerkAPIResponseError } from './resources/internal';
 
-class FraudProtectionService {
-  private inflightRequest: Promise<unknown> | null = null;
+export class FraudProtection {
+  private static instance: FraudProtection;
 
-  public async execute<T extends () => Promise<any>>(cb: T): Promise<Awaited<ReturnType<T>>> {
-    if (this.inflightRequest) {
-      await this.inflightRequest;
-    }
+  private inflightException: Promise<unknown> | null = null;
```

Comment:
```suggestion
  private inflightException: Promise<unknown> = Promise.resolve();
```

Reply:
I think we can avoid the if statements and setting this value to `null`, we can just start with a resolved promise, an `await` will work as expected on the rest of the code.

Reply:
It's not that simple, so scratch that!